File: 백준/Silver/10828. 스택/스택.js
const fs = require('fs');

const input = fs.readFileSync(0, 'utf8').trim().split('\n');
const n = Number(input[0]);

const stack = [];
const out = [];

for (let i = 1; i <= n; i++) {
  const [cmd, val] = input[i].split(' ');

  switch (cmd) {
    case 'push':
      stack.push(val);
      break;
    case 'pop':
      out.push(stack.length ? stack.pop() : -1);
      break;
    case 'size':
      out.push(stack.length);
      break;
    case 'empty':
      out.push(stack.length ? 0 : 1);
      break;
    case 'top':
      out.push(stack.length ? stack[stack.length - 1] : -1);
      break;
  }
}

process.stdout.write(out.join('\n'));

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
#include <string>
using namespace std;

int main() {
    int n;
    cin >> n;

    stack<int> stk;

    while (n--) {
        string cmd;
        cin >> cmd;

        if (cmd == "push") {
            int num;
            cin >> num;
            stk.push(num);
        }
        else if (cmd == "pop") {
            if (stk.empty()) {
                cout << "-1\n";
            }
            else {
                cout << stk.top() << '\n';
                stk.pop();
            }
        }
        else if (cmd == "size") {
            cout << stk.size() << '\n';
        }
        else if (cmd == "empty") {
            cout << (stk.empty() ? 1 : 0) << '\n';
        }
        else if (cmd == "top") {
            if (stk.empty()) {
                cout << "-1\n";
            }
            else {
                cout << stk.top() << '\n';
            }
        }
    }

    return 0;
}